### PR TITLE
fix: return empty array instead of null

### PR DESCRIPTION
**Pull request by notanatol**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md)
- [x] My change requires a documentation update and I have done it

### Description
our collection responses are not aligned when it comes to empty arrays - we get mixed `[]` and `null`

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2821)

### CODE REVIEW

**Review by acud — APPROVED** [review]

**Comment by acud** [comment]
just worth to make sure that this is reflected in the openapi spec

**Comment by AuHau** [comment]
Awesome that there is work on this! I have to remarks though.

First of all, just through searching of the OpenAPI spec, I have found 9 places where this is a problem, yet this PR tackles only three. I would suggest tackling all of those.

Second of all, be aware that this is a breaking change for the API, so please signal that with a correct bump of OpenAPI versions.

**Review by mrekucci — COMMENTED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2821)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @AuHau)

___
*[openapi/SwarmCommon.yaml, line 532 at r1](https://reviewable.io/reviews/ethersphere/bee/2821#-MwRiAA9BhSoW2APRt22:-MwRiAA9BhSoW2APRt23:b-jm42q) ([raw file](https://github.com/ethersphere/bee/blob/fe658d3adb9d97737efafde960fcd45474f422e9/openapi/SwarmCommon.yaml#L532)):*
> ```YAML
>       type: object
>       properties:
>         addresses:
> ```

This should be also renamed to `references` in order to reflect the real response.
___
*[pkg/api/pin.go, line 131 at r1](https://reviewable.io/reviews/ethersphere/bee/2821#-MwRcxLDCXyqkSqrjd0r:-MwRcxLDCXyqkSqrjd0s:blcdwh4) ([raw file](https://github.com/ethersphere/bee/blob/fe658d3adb9d97737efafde960fcd45474f422e9/pkg/api/pin.go#L131)):*
> ```Go
> 	}
> 
> 	if pinned == nil {
> ```

This doesn't make much sense, because the `Pins()` method already returns an empty array if there was no error.
___
*[pkg/debugapi/peer.go, line 104 at r1](https://reviewable.io/reviews/ethersphere/bee/2821#-MwRdusK8coP4fqpBqV7:-MwRdusLE9S9uOAzGpk7:b-fzave9) ([raw file](https://github.com/ethersphere/bee/blob/fe658d3adb9d97737efafde960fcd45474f422e9/pkg/debugapi/peer.go#L104)):*
> ```Go
> 
> func mapPeers(peers []p2p.Peer) (out []Peer) {
> 	out = make([]Peer, 0)
> ```

This can be optimized by specifying the `len(peers)` in the `make` call.
___
*[pkg/debugapi/postage.go, line 188 at r1](https://reviewable.io/reviews/ethersphere/bee/2821#-MwRezv0-j7ieGmi5Hy4:-MwRezv0-j7ieGmi5Hy5:b-6soam7) ([raw file](https://github.com/ethersphere/bee/blob/fe658d3adb9d97737efafde960fcd45474f422e9/pkg/debugapi/postage.go#L188)):*
> ```Go
> 	}
> 	if resp.Stamps == nil {
> 		resp.Stamps = make([]postageStampResponse, 0)
> ```

The stamps can be pre-allocated at beginning with `len(s.post.StampIssuers())`.

**Review by notanatol — COMMENTED** [review]
version bumped!
I made sure all the other `nullable` responses use empty arrays instead of `null`.
Also, `/batches` endpoint was returning a bare array, not it's wrapped in an object as described by the spec.

> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2821#-:-MwSS4gBBltF8WMMXFmv:b-s9iapz)* status: 0 of 7 files reviewed, all discussions resolved (waiting on @AuHau and @mrekucci)

___
*[pkg/api/pin.go, line 131 at r1](https://reviewable.io/reviews/ethersphere/bee/2821#-MwRcxLDCXyqkSqrjd0r:-MwRqTzSBXebE5U0hz-5:b-rhqxct) ([raw file](https://github.com/ethersphere/bee/blob/fe658d3adb9d97737efafde960fcd45474f422e9/pkg/api/pin.go#L131)):*
<details><summary><i>Previously, mrekucci wrote…</i></summary><blockquote>

This doesn't make much sense, because the `Pins()` method already returns an empty array if there was no error.
</blockquote></details>

I don't know, in `pinning.go:142` there only declaration, but it's not allocated
but point taken, I can move it there

**Review by mrekucci — APPROVED** [review]
> Reviewed 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2821)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @AuHau)

**Comment by AuHau** [comment]
> I made sure all the other nullable responses use empty arrays instead of null.

Sweet!

> version bumped!

Well, since it is a breaking change I would argue the major version of the versions should be bumped instead of patch version.

**Review by notanatol — COMMENTED** [review]
bumped now

> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2821#-:-MwT15QP6YhQm9YX67e6:b14crps)* status: 3 of 7 files reviewed, all discussions resolved (waiting on @AuHau and @mrekucci)

File: openapi/SwarmDebug.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 openapi: 3.0.3
 info:
-  version: 1.2.1
+  version: 2.2.1
```
Comment (AuHau) [inline openapi/SwarmDebug.yaml:3]:
```suggestion
  version: 2.0.0
```

"higher-level" version bump resets the "lower-level" versions ;-)

**Review by AuHau — CHANGES_REQUESTED** [review]

**Review by AuHau — APPROVED** [review]
LGTM! Thanks

**Pull request merged**